--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,6 +46,11 @@ validpgpkeys=(
               'DE9452CE46F42094907F108B44D1C0F82525FE49' # Bradley Spengler
 )
 
+# grsec-selinux custom definitions
+pkgbase=linux-grsec-selinux
+source+=('usr-bin-usermode-helper.patch')
+sha256sums+=('SKIP')
+
 _kernelname=${pkgbase#linux}
 
 prepare() {
@@ -66,6 +71,9 @@ prepare() {
   patch -Np1 -i "$srcdir/$_grsec_patch"
   rm localversion-grsec
 
+  # Add custom patch
+  patch -Np1 -i "$srcdir/usr-bin-usermode-helper.patch"
+
   if [ "${CARCH}" = "x86_64" ]; then
     cat "${srcdir}/config.x86_64" > ./.config
   else
