physmem_size is defined as unsigned long long elsewhere

--- a/arch/um/kernel/um_arch.c
+++ b/arch/um/kernel/um_arch.c
@@ -121,7 +121,7 @@ int ncpus = 1;
 static int have_root __initdata = 0;
 
 /* Set in uml_mem_setup and modified in linux_main */
-long long physmem_size = 32 * 1024 * 1024;
+unsigned long long physmem_size = 32 * 1024 * 1024;
 
 static const char *usage_string =
 "User Mode Linux v%s\n"
